### ISSUE TITLE
The option "Do not send unsubscription link" does not work anymore

### ISSUE THREAD

**Issue by dasistwas** [Wunderbyte-GmbH/moodle-mod_newsletter#46]
Reproduce

1) Go to manage subscriptions
2) Edit a single subscription
3) Activate checkbox "Do not send unsubscribe link"
4) Create an issue
5) Deliver issue
6) Result: Unsubscribe link is sent, but should not be sent.

Fix: Do not send unsubscribe link when checkbox is activated.

**Comment by michaelpollak** [comment]
I have tested with mdl 3.8 and 3.5, works as expected here. Can you send me some more details?

**Comment by dasistwas** [comment]
Could you try with attached HTML. I suspect parsing is not successful when complicated HTML including other links is used:
@michaelpollak 
[newsletter-content.txt](https://github.com/Wunderbyte-GmbH/moodle-mod_newsletter/files/5684752/newsletter-content.txt)

**Comment by michaelpollak** [comment]
Tested with the newsletter you posted, works the same way for me. Users that do have the checkbox ticked to "not send unsubscribe links" see the newsletter without the last line.
If debug is activated you can see some messages in the scheduled tasks interface, maybe this can lead to some more information. For example a user that will not receive an unsub link is logged this way:

```
Processing newsletter (id = 1), issue "complicated newsletter" (id = 37)...
Sending message to test@michaelpollak.org... 
Sending no unsublink to test@michaelpollak.org for 37
```

**Comment by dasistwas** [comment]
Logging is one thing. But that does not mean that the link is actually removed. Only way to find out is the control the inbox of the receiver. My suspicion is that the regex for replacing fails.

**Comment by michaelpollak** [comment]
I agree, but I also tested it with sent mails and got no unsubscribe link when appropriate. It would be interesting if the problematic account is logged correctly - then the bug could be in the regex - or not which would indicate a problem in the user data or somewhere where we fetch this information from the database.
